tests/test_views.py

```diff
@@ -303,18 +304,18 @@ def test_env_entry_render_params(bukudb, app, client, mode, favicons, new_tab):
         app.config.update({'BUKUSERVER_OPEN_IN_NEW_TAB': new_tab})
 
     dom = assert_response(client.get('/bookmark/'), '/bookmark/')
-    cell = ' '.join(etree.tostring(dom.xpath('//td[@class="col-Entry"]')[0], encoding='unicode').strip().split())
+    cell = ' '.join(etree.tostring(dom.xpath(f'//td{xpath_cls("col-Entry")}')[0], encoding='unicode').strip().split())
     target = '' if not new_tab else ' target="_blank"'
-    icon = '' if not favicons else f'<img src="http://www.google.com/s2/favicons?domain={netloc}"/> '
-    prefix = f'<td class="col-Entry"> {icon}<a href="{url}"{target}>{title}</a>'
-    suffix = (''.join(f'<a class="btn btn-default" href="/bookmark/?flt0_tags_contain={s}">{s}</a>' for s in _tags) +
-              f'<br/>{desc} </td>')
+    icon = '' if not favicons else f'<img class="favicon" src="http://www.google.com/s2/favicons?domain={netloc}"/> '
+    prefix = f'<td class="col-Entry"> {icon}<span class="title"><a href="{url}"{target}>{title}</a></span>'
+    tags = [f'<a class="btn label label-default" href="/bookmark/?flt0_tags_contain={s}">{s}</a>' for s in _tags]
+    netloc_tag = ('' if mode == 'netloc' else
+                  f'<a class="btn label label-success" href="/bookmark/?flt0_url_netloc_match={netloc}">netloc:{netloc}</a>')
+    suffix = f'<div class="tag-list">{netloc_tag}{"".join(tags)}</div><div class="description">{desc}</div> </td>'
     if mode == 'netloc':
-        assert cell == prefix + f' (<a href="/bookmark/?flt0_url_netloc_match={netloc}">{netloc}</a>)<br/>' + suffix
+        assert cell == f'{prefix}<span class="netloc"> (<a href="/bookmark/?flt0_url_netloc_match={netloc}">{netloc}</a>)</span>{suffix}'
     else:
-        assert cell == (prefix + f'<br/><a href="{url}"{target}>{url}</a><br/>' +
-                        f'<a class="btn btn-default" href="/bookmark/?flt0_url_netloc_match={netloc}">netloc:{netloc}</a>' +
-                        suffix)
+        assert cell == f'{prefix}<span class="link"><a href="{url}"{target}>{url}</a></span>{suffix}'
```

Comment:
Note that there's no `<br/>`s now; tags and description are in `<div>`s instead, and `.link` has `display: block`.

Also, every component has its own DOM node (tagged with an appropriate CSS class), allowing for further CSS-only formatting.